ewmarket.py

```diff
@@ -1008,3 +1008,237 @@ def get_majority_shareholder(id_server = None, stock = None):
+
+			resp_cont.add_channel_response(cmd.message.channel.name, user_offers)
+			resp_cont.add_channel_response(cmd.message.channel.name, partner_offers)
+			resp_cont.format_channel_response(cmd.message.channel.name, cmd.message.author)
+
+			await resp_cont.post()
+
+	else:
+		if cmd.mentions_count == 0:
+			return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, "Who do you want to trade with?"))
+		
+		if cmd.mentions_count > 1: 
+			return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, "You can only trade with one person at a time."))
+
+		trade_partner = EwUser(member=cmd.mentions[0])
+
+		if user_data.id_user == trade_partner.id_user:
+			return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, "Huh?"))
+
+		if user_data.poi != trade_partner.poi:
+			return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, "You must be in the same location as someone to trade with them."))
+		
+		if ewutils.active_trades.get(trade_partner.id_user) != None and len(ewutils.active_trades.get(trade_partner.id_user)) > 0:
+			return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, "Wait for them to finish their business before trying to trade with them."))
+
+		ewutils.active_trades[user_data.id_user] = {"state": ewcfg.trade_state_proposed, "trader": trade_partner.id_user}
+		ewutils.active_trades[trade_partner.id_user] = {"state": ewcfg.trade_state_proposed, "trader": user_data.id_user}
+
+		await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.mentions[0], "{user} wants to trade with you. Do you {accept} or {refuse}?".format(user=cmd.message.author.display_name, accept=ewcfg.cmd_accept, refuse=ewcfg.cmd_refuse)))
+
+		accepted = False
+
+		try:
+			member = cmd.mentions[0]
+			msg = await cmd.client.wait_for_message(timeout = 30, author = member, check = ewutils.check_accept_or_refuse)
+
+			if msg != None and msg.content.lower() == ewcfg.cmd_accept:
+				accepted = True
+		except:
+			accepted = False
+
+		if accepted:
+			ewutils.active_trades[user_data.id_user] = {"state": ewcfg.trade_state_ongoing, "trader": trade_partner.id_user}
+			ewutils.active_trades[trade_partner.id_user] = {"state": ewcfg.trade_state_ongoing, "trader": user_data.id_user}
+
+			ewutils.trading_offers[user_data.id_user] = []
+
+			ewutils.trading_offers[trade_partner.id_user] = []
+
+			response = "You both head into a shady allyway nearby to conduct your business."
+
+			await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+
+		else:
+			ewutils.active_trades[user_data.id_user] = {}
+			ewutils.active_trades[trade_partner.id_user] = {}
+			print("timwout")
```

Comment:
looks like a debug thing. maybe have EW send confirmation that the trade has been rejected instead